app/models/liabilitycalculation/viewmodels/TaxDueSummaryViewModel.scala

```diff
@@ -125,10 +125,29 @@ object TaxDueSummaryViewModel {
         totalAnnuityPaymentsTaxCharged = calc.taxCalculation.flatMap(tc => tc.totalAnnuityPaymentsTaxCharged),
         totalRoyaltyPaymentsTaxCharged = calc.taxCalculation.flatMap(tc => tc.totalRoyaltyPaymentsTaxCharged),
         giftAidTaxChargeWhereBasicRateDiffers = calc.taxCalculation.flatMap(tc => tc.incomeTax.giftAidTaxChargeWhereBasicRateDiffers),
-        totalTaxableTransitionProfit = calc.transitionProfit.flatMap(_.totalTaxableTransitionProfit),
-        incomeTaxChargedOnTransitionProfits = calc.taxCalculation.flatMap(_.incomeTax.incomeTaxChargedOnTransitionProfits)
+        transitionProfitRow = TransitionProfitRow(
+          calc.taxCalculation.flatMap(_.incomeTax.incomeTaxChargedOnTransitionProfits),
+          calc.transitionProfit.flatMap(_.totalTaxableTransitionProfit))
       )
       case None => TaxDueSummaryViewModel()
     }
   }
 }
+
+object TransitionProfitRow {
+  def apply(incomeTaxCharged: Option[BigDecimal], totalTaxableProfit: Option[BigDecimal]): Option[TransitionProfitRow] = {
+    (incomeTaxCharged, totalTaxableProfit) match {
```

Comment:
Did you consider doing this with flatMap()? I think it could be a 1-liner.

Reply:
Keep logs will provide as a clue in case of missing fields

Reply:
Yeah. As discussed in the Tech Deep Dive, I wish we didn't need to do this, but let's keep them.